src/core.js

```diff
@@ -29,7 +29,8 @@ var plotApi = require('./plot_api');
 var methodNames = Object.keys(plotApi);
 for(var i = 0; i < methodNames.length; i++) {
     var name = methodNames[i];
-    exports[name] = plotApi[name];
+    // _ -> private API methods, but still registered for internal use
+    if(name.charAt(0) !== '_') exports[name] = plotApi[name];
```

Comment:
For this PR I chose to draw a distinction between GUI-driven changes - which we record so they can be persisted later - and other random `restyle` and `relayout` calls. My thinking was that other calls are explicitly being invoked by the developer, so they can also decide for themselves whether to persist them by taking into account the accompanying setting when the figure is next updated.

This line prevents exposing these variants (as `Plotly._guiRelayout` etc) but I guess I could see an argument for allowing app developers to access them. Say for example they want to extend our modebar (or make other similar controls) with things like a log/linear axis control. If they did this with `_guiRelayout` it could be all self-contained in the plot widget rather than tied into the higher-level logic about generating the figure. That said in this particular case it could lead to subtle problems since the form of `axis.range` is different for log and linear axes, so a figure that doesn't know what kind of axis it's plotting onto could supply a faulty explicit range...

So after all that I chose not to expose this to app developers, but I'm happy to discuss.